TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/IScopedInstrumentationAttributes.java

```diff
@@ -0,0 +1,44 @@
+package org.opensearch.migrations.tracing;
+
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.trace.Span;
+import lombok.NonNull;
+
+public interface IScopedInstrumentationAttributes
+        extends IWithStartTimeAndAttributes, AutoCloseable {
+
+    String getActivityName();
+
+    @Override
+    @NonNull Span getCurrentSpan();
+    CommonScopedMetricInstruments getMetrics();
+    default LongCounter getEndOfScopeCountMetric() {
+        return getMetrics().contextCounter;
+    }
+    default DoubleHistogram getEndOfScopeDurationMetric() {
+        return getMetrics().contextDuration;
+    }
+    default void endSpan() {
+        getCurrentSpan().end();
+    }
+
+    default void sendMeterEventsForEnd() {
+        meterIncrementEvent(getEndOfScopeCountMetric());
+        meterHistogramMillis(getEndOfScopeDurationMetric());
+    }
+
+    default void close() {
+        endSpan();
+        sendMeterEventsForEnd();
+    }
+
+    default void addException(Throwable e) {
```

Comment:
Should this add a call to setObservedExceptionToIncludeInMetrics or can setObservedExceptionToIncludeInMetrics and associated functions be removed in favor of the exceptionCounter

Reply:
I had an older version pulled locally. IInstrumentationAttributes has been changed to remove setObservedExceptionToIncludeInMetrics from IInstrumentationAttributes